### PR TITLE
[receiver/k8scluster] Change k8s.cronjob.active_jobs to use mdatagen

### PR DESCRIPTION
**Pull request by povilasv**
**Description:** Change k8s.cronjob.active_jobs to use mdatagen

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4367

### CODE REVIEW

**Comment by dmitryax** [comment]
Same comment as in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24055

File: receiver/k8sclusterreceiver/internal/cronjob/documentation.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The following metrics are emitted by default. Each of them can be disabled by applying the following configuration:
+
+```yaml
+metrics:
+  <metric_name>:
+    enabled: false
+```
+
+### k8s.cronjob.active_job
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/internal/cronjob/documentation.md:17]: It's called `k8s.cronjob.active_jobs` not `k8s.cronjob.active_job` :)

File: receiver/k8sclusterreceiver/internal/cronjob/cronjobs.go [hunk 2]
```diff
@@ -23,71 +24,24 @@ const (
...
+	mbphase.RecordK8sCronjobActiveJobDataPoint(ts, int64(len(cj.Status.Active)))
 
-	return []*agentmetricspb.ExportMetricsServiceRequest{
-		{
-			Resource: getResourceForCronJob(cj),
-			Metrics:  metrics,
-		},
-	}
+	metrics := mbphase.Emit(imetadataphase.WithK8sNamespaceName(cj.Namespace), imetadataphase.WithK8sCronjobUID(string(cj.UID)), imetadataphase.WithK8sCronjobName(cj.Name), imetadataphase.WithOpencensusResourcetype("k8s"))
+	return metrics
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/internal/cronjob/cronjobs.go:34]:
```suggestion
	return mbphase.Emit(imetadataphase.WithK8sNamespaceName(cj.Namespace), imetadataphase.WithK8sCronjobUID(string(cj.UID)), imetadataphase.WithK8sCronjobName(cj.Name), imetadataphase.WithOpencensusResourcetype("k8s"))
```

Comment (atoulme) [reply receiver/k8sclusterreceiver/internal/cronjob/cronjobs.go:34]: Same change line 43

File: receiver/k8sclusterreceiver/internal/cronjob/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+type: k8s/cronjob
+
+sem_conv_version: 1.18.0
+
+parent: k8s_cluster
+
+resource_attributes:
+  k8s.cronjob.uid:
+    description: THe k8s CronJob uid.
```
Comment (atoulme) [inline receiver/k8sclusterreceiver/internal/cronjob/metadata.yaml:9]:
```suggestion
    description: The k8s CronJob uid.
```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**